src/pg/sql/41_observatory_augmentation.sql

```diff
@@ -1076,3 +1076,49 @@ BEGIN
   RETURN result;
 END;
 $$ LANGUAGE plpgsql STABLE;
+
+-- MetadataValidation checks the metadata parameters and the geometry type
+-- of the data in order to find possible wrong cases
+CREATE OR REPLACE FUNCTION cdb_observatory.obs_metadatavalidation(
+  geometry_extent geometry(Geometry, 4326),
+  geometry_type text,
+  params JSON,
+  target_geoms INTEGER DEFAULT NULL
+)
+RETURNS TABLE(valid boolean, errors text[]) AS $$
+DECLARE
+  meta_query text;
+  meta json;
+  metadata_result json;
+  errors text[];
+BEGIN
+  errors := (ARRAY[])::TEXT[];
+  IF geometry_type IN ('ST_Polygon', 'ST_MultiPolygon') THEN
+      EXECUTE 'SELECT cdb_observatory.OBS_GetMeta($1, $2, 1, 1, $3)' INTO metadata_result USING geometry_extent, params, target_geoms;
+      FOR meta IN SELECT * FROM json_array_elements(metadata_result)
```

Comment:
PL/pgSQL newbie question. Could we not move `json_array_elements` to the query above?

```
EXECUTE 'SELECT json_array_elements(cdb_observatory.OBS_GetMeta($1, $2, 1, 1, $3))' INTO metadata_result USING geometry_extent, params, target_geoms;
FOR meta IN metadata_result
```

We would have to change the type of `metadata_result` to be a recordset, which I don't know if it is a valid type for variables.